plugins/evaluation-plugin/src/com/intellij/cce/evaluable/rename/RenameActionsInvoker.kt

```diff
@@ -0,0 +1,66 @@
+// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package com.intellij.cce.evaluable.rename
+
+import com.intellij.cce.core.*
+import com.intellij.cce.evaluable.common.BasicActionsInvoker
+import com.intellij.codeInsight.lookup.LookupManager
+import com.intellij.codeInsight.lookup.impl.LookupImpl
+import com.intellij.completion.ml.actions.MLCompletionFeaturesUtil
+import com.intellij.ide.DataManager
+import com.intellij.openapi.actionSystem.ActionManager
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.Presentation
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.refactoring.actions.RenameElementAction
+import com.intellij.refactoring.rename.inplace.InplaceRefactoring
+
+class RenameActionsInvoker(private val project: Project,
+                           private val language: Language,
+                           private val strategy: RenameStrategy) : BasicActionsInvoker(project, language) {
+
+  override fun callFeature(expectedText: String, prefix: String?): Lookup {
+    LOG.info("Call rename. Expected text: $expectedText. ${positionToString(editor!!.caretModel.offset)}")
+    //        assert(!dumbService.isDumb) { "Calling completion during indexing." }
+
+    val start = System.currentTimeMillis()
+
+    val myEditor = editor
```

Comment:
can throw exception if editor is null: `val editor = editor ?: throw IllegalStateException("No open editor")`